ecc/src/base_field_ecc.rs

```diff
@@ -42,33 +40,27 @@ impl<C: CurveAffine, const NUMBER_OF_LIMBS: usize, const BIT_LEN_LIMB: usize>
 {
     /// Return `BaseEccChip` from `EccConfig`
     pub fn new(config: EccConfig) -> Self {
-        let rns = Self::rns();
         Self {
-            config,
-            rns: Rc::new(rns),
+            integer_chip: IntegerChip::new(config.integer_chip_config(), Rc::new(Rns::construct())),
```

Comment:
I checked this a big and observed two things
1. Rns is a big struct.  I checked it's size with the following snippet and it's 1080 bytes:
```rust
#[cfg(test)]
mod tests {
    use super::*;
    use crate::curves::bn256::Fr as BnScalar;
    #[test]
    fn rns_size() {
        println!(
            "Rns size: {:?}",
            std::mem::size_of::<Rns<BnScalar, BnScalar, 4, 68>>()
        );
    }
}
```
2. The `Rns` appears in many "lightweight" values like the `Integer`.  A circuit may have lots of `Integer` values, and requiring that each one keeps the exact same copy of the `rns` value may be bad for memory (and also implies lots of copying)
3. Probably using a reference to a shared `rns` would be annoying because then all the Integers depend on an `rns` that must survive their lifetime.

So I find it reasonable to use `Rc` here to avoid having a compile-time lifetime analysis of the `rns` value that is shared among many values; which simplifies the code and I think doesn't add any penalty.

Reply:
Agreed. After the discussions we had. I completelly agree that this is the best way.

Thanks for the help on that @ed255